src/app/portlets/content-types/content-types-resolver.service.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { Observable } from 'rxjs/Observable';
+import { LoginServiceMock } from '../../test/login-service.mock';
+import { RouterTestingModule } from '@angular/router/testing';
+import { inject } from '@angular/core/testing';
+import { DotRouterService } from '../../api/services/dot-router-service';
+
+class CrudServiceMock {
+    getDataById() {}
+}
+
+class DotRouterServiceMock {
+    gotoPortlet() {}
+}
+
+const activatedRouteSnapshotMock: any = jasmine.createSpyObj<
+    ActivatedRouteSnapshot
+>('ActivatedRouteSnapshot', ['toString']);
+activatedRouteSnapshotMock.paramMap = {};
+
+describe('ContentTypeResolver', () => {
+    let crudService: CrudService;
+    let router: ActivatedRouteSnapshot;
+    let contentTypeResolver: ContentTypeResolver;
+    let dotRouterService: DotRouterService;
+
+    beforeEach(
+        async(() => {
+            TestBed.configureTestingModule({
+                providers: [
+                    ContentTypeResolver,
+                    ContentTypesInfoService,
+                    {
+                        provide: DotRouterService,
+                        useClass: DotRouterServiceMock
+                    },
+                    { provide: CrudService, useClass: CrudServiceMock },
+                    { provide: LoginService, useClass: LoginServiceMock },
+                    {
+                        provide: ActivatedRouteSnapshot,
+                        useValue: activatedRouteSnapshotMock
+                    }
+                ],
+                imports: [RouterTestingModule]
+            });
+            crudService = TestBed.get(CrudService);
+            router = TestBed.get(ActivatedRouteSnapshot);
+            contentTypeResolver = TestBed.get(ContentTypeResolver);
+            dotRouterService = TestBed.get(DotRouterService);
+        })
+    );
+
+    it('should get and return a content type', () => {
+        activatedRouteSnapshotMock.paramMap.get = () => '123';
+        spyOn(crudService, 'getDataById').and.returnValue(Observable.of({
+            fake: 'content-type',
+            object: 'right?'
+        }));
+        contentTypeResolver
+            .resolve(activatedRouteSnapshotMock)
+            .subscribe((fakeContentType: any) => {
+                expect(fakeContentType).toEqual({
+                    fake: 'content-type',
+                    object: 'right?'
+                });
+            });
+        expect(crudService.getDataById).toHaveBeenCalledWith(
+            'v1/contenttype',
+            '123'
+        );
+    });
+
+    it('should get and return null and go to home', () => {
+        activatedRouteSnapshotMock.paramMap.get = () => '123';
+        spyOn(dotRouterService, 'gotoPortlet');
+        spyOn(crudService, 'getDataById').and.returnValue(Observable.of(false));
+        contentTypeResolver
+            .resolve(activatedRouteSnapshotMock)
+            .subscribe((res: any) => {
+                expect(res).toBeNull();
```

Comment:
are you sure that this expect always is executed?, I think you need a tick or a whenStable o something like that

Reply:
So here is the deal, because the ```resolve``` is async also the method he use internally... but we are mocking all the responses and they are coming inmediatelly, we could use ```fakeAsync``` but it's not necessary because all the responses are inmediatelly available when the subscribe it's called.